default/python/AI/CombatRatingsAI/_ratings.py

```diff
@@ -49,10 +50,17 @@ def default_ship_stats() -> ShipCombatStats:
     )
 
 
-def weight_shields(shields: float, grade: str) -> float:
-    """Re-weights shields based on species defense bonus."""
-    offset = {"NO": 0, "BAD": 0, "": 0, "GOOD": 1.0, "GREAT": 0, "ULTIMATE": 0}.get(grade, 0)
-    return shields + offset * AIDependencies.SHIP_SHIELD_FACTOR
+def species_shield_bonus(species: Optional[SpeciesName], shield_type: Optional[str]) -> float:
+    skill = get_species_ship_shields(species)
```

Comment:
I think all those get_species_-functions should also have argument type Optional[SpeciesName], but that's for another PR.